tests/RHF/Constrained_RHF_test.cpp

```diff
@@ -80,7 +80,105 @@ BOOST_AUTO_TEST_CASE ( constrained_CO_test) {
         // Contrain the original Hamiltonian parameters
         auto constrained_ham_par = ao_ham_par.constrain(mulliken_operator, i);
 
-        // Create a plain RHF SCF solver and solve the SCF equations
+        // Create a DIIS RHF SCF solver and solve the SCF equations
+        GQCP::DIISRHFSCFSolver diis_scf_solver (constrained_ham_par, CO, 6, 10e-12, 10000);
+        diis_scf_solver.solve();
+        auto rhf = diis_scf_solver.get_solution();
+
+        // Transform the mulliken operator to the basis in which the RHF energies are calculated
+        mulliken_operator.transform(rhf.get_C());
+
+        // Retrieve the RHF "energy"
+        double expectation_value = rhf.get_electronic_energy();
+
+        // Retrieve the expectation value of the Mulliken operator (aka the population)
+        double mulliken_population = GQCP::calculateExpectationValue(mulliken_operator, one_rdm);
+
+        // Retrieve the total energy by adding the lambda times the expectation value of the constraining operator
+        double total_energy = expectation_value + i * mulliken_population + CO.calculateInternuclearRepulsionEnergy();
+
+        // Mulliken charge on the carbon atom
+        double C_charge = 6 - mulliken_population;
+
+        BOOST_CHECK(std::abs(total_energy - CO_data(index, 2)) < 1.0e-4);
+        BOOST_CHECK(std::abs(C_charge - CO_data(index, 1)) < 1.0e-2);
+
+        index++;
+    }
+}
+
+BOOST_AUTO_TEST_CASE ( constrained_CO_test_random_transformation) {
+    // Repeat the same test but perform a random transformation
```

Comment:
Document as to why this test is necessary. Add one whiteline before, remove some after